src/steamship/agents/logging.py

```diff
@@ -37,3 +42,34 @@ class AgentLogging:
     AGENT = "agent"
     TOOL = "tool"
     SYSTEM = "system"
+    LLM = "llm"
+
+
+LOGGING_FORMAT = {
+    "level": "%(levelname)s",
+    "host": "%(hostname)s",
+    "where": "%(module)s.%(filename)s.%(funcName)s:%(lineno)s",
+    "type": "%(levelname)s",
+    "stack_trace": "%(exc_text)s",
+    "message_type": "%(message_type)s",
+    "component_name": "%(component_name)s",
+    AgentLogging.IS_MESSAGE: f"%({AgentLogging.IS_MESSAGE})s",  # b doesn't work. Unsure how to make a bool
+    AgentLogging.AGENT_NAME: f"%({AgentLogging.AGENT_NAME})s",
+    AgentLogging.MESSAGE_AUTHOR: f"%({AgentLogging.MESSAGE_AUTHOR})s",
+    AgentLogging.MESSAGE_TYPE: f"%({AgentLogging.MESSAGE_TYPE})s",
+    AgentLogging.TOOL_NAME: f"%({AgentLogging.TOOL_NAME})s",
+}
+
+
+class StreamingOpts(BaseModel):
```

Comment:
I like how you've started with the idea that what gets added to the stream can be selected.

One thing that would help me (and therefor I think others?) is to actively add comments here about what distinguishes an `agent` and `llm` and `tool` message.

For me: I find myself wanting to be reassured about my guess about `agent` versus `llm` specifically.

For ex:
- Is `agent_message` something the agent says that gets added to the chat history?
- Is `llm_message` something that the LLM emits to a tool or agent, but doesn't tend to get added to the chat history?

Reply:
Added some docs. One thing that I wanted to make more distinct in this PR is the boundary between `Agent` and `LLM`. There is nothing in `Agent` that requires `LLM` integration or backing. You could have a purely random tool selection agent, for instance. So, I wanted to allow for `LLM`s to be responsible for `LLM`-scoped messages and `Agent`s to focus on the level above in logs. This might come at a cost of a bit more of verbosity, however.